scripts/run_frontend_tests.py

```diff
@@ -89,14 +89,24 @@ def main(args=None):
             'start', os.path.join('core', 'tests', 'karma.conf.ts')]
 
     task = subprocess.Popen(cmd, stdout=subprocess.PIPE)
-    out, _ = task.communicate()
-    task.wait()
+    output_lines = []
+    # Reads and prints realtime output from the subprocess until it terminates.
+    while True:
+        line = task.stdout.readline()
```

Comment:
![image](https://user-images.githubusercontent.com/16653571/85608916-2e26a400-b673-11ea-8f48-2748d4229275.png)
[ref](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.stderr)

Should we avoid reading a line from stdout? (not sure)

Reply:
Just double checked that, it should be an issue only when we are trying to use `.stdin.write` followed by `.stdout.read`. In our case, we are not writing to `.stdin` and are actively consuming from `.stdout`, so it should not be an issue.

Reply:
As per the "Warning" it's about **other proccess** can lead this into a deadlock if we read/write through any of the  in/out/err pipe (not  .stdin.write followed by .stdout.read), right?

Reply:
I think the intention of this warning is to tell us to be careful while using `.stdin` `.stdout`, not to prohibit the usage of any of them.
In our case, parent process is blocked on `.stdout.readline()` only when the subprocess' `stdout` buffer is empty, and the subprocess is blocked only when its `stdout` buffer is full. So in my opinion they will not be blocked simutaneously, no circular wait.

The other case (`.stdin.write` followed by `.stdout.read`) can lead to a deadlock. Parent process is blocked while writing to a full `stdin` buffer. If at the same time, the subprocess is not reading from its `stdin`, but writing to its `stdout`, which is full as well, the deadlock appears.

Reply:
@MegrezZhu I had a discussion on this with other members and we thought of living this as -is and adjust it in future if needed. (Sorry for being late!)